pkg/apis/scheduling/v1alpha1/types_location.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Location represents a scheduling target for objects of a given type.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories=kcp
+// +kubebuilder:printcolumn:name="Type",type=string,JSONPath=`.spec.type`,description="Type of the workspace"
+// +kubebuilder:printcolumn:name="Phase",type=string,JSONPath=`.status.phase`,description="The current phase (e.g. Scheduling, Initializing, Ready)"
+// +kubebuilder:printcolumn:name="URL",type=string,JSONPath=`.status.baseURL`,description="URL to access the workspace"
+type Location struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec LocationSpec `json:"spec,omitempty"`
+
+	// +optional
+	Status LocationStatus `json:"status,omitempty"`
+}
+
+// LocationPoolReference defines a reference to a location pool.
```

Comment:
can user also create a location without specifying location pool? It seems like a locationPool is to define common properties of locations. So there will be a controller to generate location by location pool, and there should be another controller to bind a location to an instance.

Reply:
No, the user cannot define Locations. The LocationDomain owner defines which locations exists. A location is a logically sensible partition of the set of workload cluster. E.g. a region of a cloud provider.